### PR TITLE
resource as query param generic endpoint

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Check if indexes are used
- Optimize jsonb queries 

## Related Issue(s)
- #316 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added:PR #404 
- [x] All tests run green

### CODE REVIEW

File: src/Events/Controllers/EventsController.cs [hunk 1]
```diff
@@ -99,8 +100,8 @@ public async Task<ActionResult<string>> Post([FromBody] CloudEvent cloudEvent)
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [Produces("application/cloudevents+json")]
         public async Task<ActionResult<List<CloudEvent>>> Get(
+            [FromQuery, Required] string resource,
```
Comment (SandGrainOne) [inline src/Events/Controllers/EventsController.cs:103]: What happens if the parameter is missing? Will the response be 400 Bad Request or will it be 404 Not Found?

File: src/Events/Swagger/RequestBodyExampleFilter.cs [hunk 2]
```diff
@@ -41,6 +41,8 @@ private static void CreateCloudEventRequestModelExamples(OpenApiRequestBody requ
                 ("Instance created event with alternative subject",
                 CreateOpenApiObject(new List<(string Name, string Value)>()
                 {
+                    ("resource", "urn:altinn:resource:altinnapp.ttd.apps-test"),
```
Comment (SandGrainOne) [inline src/Events/Swagger/RequestBodyExampleFilter.cs:44]: Should we update the example for the Generic POST event endpoint as well?

Comment (acn-sbuad) [reply src/Events/Swagger/RequestBodyExampleFilter.cs:44]: the generic post example is based on the cloud event schema. Added resource prop to that schema as well. We don't have a set of example request bodies as we have for other endpoints.

File: src/Events/Controllers/EventsController.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Events/Controllers/EventsController.cs:103]: Good

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**